crates/sparrow-compiler/src/ast_to_dfg.rs

```diff
@@ -527,31 +527,28 @@ pub(super) fn add_to_dfg(
                 dfg.bind("$condition_input", args[0].inner().clone());
 
                 let window = &expr.unwrap().args()[1];
-                let (condition, duration) = match window.op() {
-                    ExprOp::Call(window_name) => {
-                        flatten_window_args(window_name, window, dfg, data_context, diagnostics)?
-                    }
-                    ExprOp::Literal(v) if v.inner() == &LiteralValue::Null => {
-                        // Unwindowed aggregations just use nulls
-                        let null_arg = dfg.add_literal(LiteralValue::Null.to_scalar()?)?;
-                        let null_arg = Located::new(
-                            add_literal(
-                                dfg,
-                                null_arg,
-                                FenlType::Concrete(DataType::Null),
-                                window.location().clone(),
-                            )?,
-                            window.location().clone(),
-                        );
-
-                        (null_arg.clone(), null_arg)
-                    }
-                    unexpected => anyhow::bail!("expected window, found {:?}", unexpected),
-                };
+                let (condition, duration) =
+                    flatten_window_args_if_needed(window, dfg, data_context, diagnostics)?;
 
                 dfg.exit_env();
                 // [agg_input, condition, duration]
                 vec![args[0].clone(), condition, duration]
+            } else if function.name() == "collect" {
+                // The collect function contains a window, but does not follow the same signature
+                // pattern as aggregations, so it requires a different flattening strategy.
+                //
+                // TODO: Flattening the window arguments is hacky and confusing. We should instead
```

Comment:
+1 to eliminating the tick. I was also thinking that it would help us avoid generating all the ticks. Consider:

1. We know that we tick every hour
2. We know that the window was *previously* empty
3. Therefore, the new tick doesn't have a different value
4. Therefore, we don't need to do anything for this entity.

Vs.

1. We need to produce a tick for every hour no matter what.

Reply:
Though, we still want to produce a row for that entity at each hour even if the value is empty.